video_xblock/mixins.py

```diff
@@ -285,6 +378,45 @@ def srt_to_vtt(self, request, _suffix=''):
         caps = requests.get(request.host_url + caps_path).text
         return Response(self.convert_caps_to_vtt(caps))
 
+    @XBlock.handler
+    def fetch_from_tree_play_media(self, request, _suffix=''):
+        """
+        Proxy handler to hide real API url.
+
+        Arguments:
+            request (webob.Request): The request to handle
+            suffix (string): not used
+            query string: 'language_id=transcript_id'
+        Returns:
+            webob.Response: WebVTT transcripts wrapped in Response object.
+        """
+        lang_id, tid = request.query_string.split('=')
+        transcript = self.fetch_3pm_translation(transcript_data={'id': tid, 'language_id': lang_id})
+        return Response(transcript.content)
+
+    @XBlock.handler
+    def validate_three_play_media_config(self, request, _suffix=''):
+        """
+        Handler to validate actuality of provided API credentials.
+
+        Arguments:
+            request (webob.Request):
+            suffix (string): not used
+        Returns:
+            webob.Response: (json) {'isValid': true/false}
+        """
+        api_key = request.json.get('api_key')
+        file_id = request.json.get('file_id')
+
+        if not (api_key and file_id):
+            return Response(status=400)
+
+        results = self.get_available_3pm_transcripts(file_id, api_key)
+
+        is_valid = not isinstance(results, dict)
```

Comment:
This looks strange. Please reformulate in positive key or make it more specific.